nucypher/acumen/perception.py

```diff
@@ -168,8 +218,91 @@ def abridged_state_details(state):
                 "updated": state.updated.rfc2822(),
                 }
 
-    def mark_as(self, label: Exception, node: "Teacher"):
-        self._marked[label].append(node)
-
-        if self._nodes.get(node):
-            del self._nodes[node]
+    def __reset_node_label_tracking(self, node: "Teacher"):
+        for strategy in self.__pruning_strategies:
+            strategy(node=node, reset=True)
+
+    def register_node_pruning_strategy(self,
+                                       label,
+                                       strategy: Callable,
+                                       layer: Optional[int] = None
+                                       ) -> None:
+        if label not in self.BUCKETS:
+            raise self.UnknownLabel
```

Comment:
This will benefit from a message as is done elsewhere in this file (`f"{label} is not a prunable node bucket."`)